docs/release/release-tasks.md

```diff
@@ -0,0 +1,384 @@
+
+1. Update supported versions in versions.md.
+   <br>Prior art: [Update supported versions for v1.3](https://github.com/kubernetes-sigs/cluster-api/pull/6850)
+
+#### Ensure the book for the new release is available 
+
+The goal of this task to make the book for the current release available under `https://release-1-4.cluster-api.sigs.k8s.io`.
+
+1. Add a DNS entry for the book of the new release (should be available under `https://release-1-4.cluster-api.sigs.k8s.io`).
+   <br>Prior art: [Add DNS for CAPI release-1.2 release branch](https://github.com/kubernetes/k8s.io/pull/3872)
+2. Open `https://release-1-4.cluster-api.sigs.k8s.io/` and verify that the certificates are valid
+   If they are not, talk to someone with access to Netlify, they have to click the `renew certificate` button in the Netlify UI.
+3. Update references in introduction.md only on the main branch (drop unsupported versions, add the new release version).
+   <br>Prior art: [Add release 1.2 book link](https://github.com/kubernetes-sigs/cluster-api/pull/6697)
+
+#### Polish release notes
+
+1. Checkout the latest commit on the `release-1.4` branch.
+2. Generate release notes with `make release-notes`.
+3. Finalize the release notes:
+    1. Copy & paste the release notes into a hackmd (makes collaboration very easy).
+    2. Pay close attention to the `## :question: Sort these by hand` section, as it contains items that need to be manually sorted.
+    3. Ensure consistent formatting of entries (e.g. prefix (see [v1.2.0](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.2.0) release notes)).
+    4. Merge dependency bump PR entries for the same dependency into a single entry.
+    5. Move minor changes into a single line at the end of each section.
+    6. Write highlights section based on the initial release notes doc.
+    7. Add Kubernetes version support section.
+4. Iterate until the GA release (the release note tool can be run starting from a specific commit).
+
+#### Change production branch in Netlify to the new release branch
+
+The goal of this task to make the book for the current release available under `https://cluster-api.sigs.k8s.io`.
+
+Someone with access to Netlify should:
+1. Change production branch in Netlify to `release-1.4` to make the book available under `https://cluster-api.sigs.k8s.io`.
+2. Re-deploy via the Netlify UI.
+
+#### Update clusterctl links in the quickstart 
+
+The goal of this task is to ensure the quickstart has links to the latest `clusterctl` binaries.
+
+1. Update clusterctl links in the quickstart (on main and cherry-pick onto release-1.4).
+   <br>Prior art: [Update clusterctl version to v1.2.x in quick start](https://github.com/kubernetes-sigs/cluster-api/pull/6716)
+
+**Note**: The PR for this should be merged after the minor release has been published.
+
+#### Continuously: Communicate key dates to the community
+
+The goal of this task is to ensure all stakeholders are informed about the current release cycle.
```

Comment:
I think this can probably be made a bit more specific once we figure out a good way to do this

@oscr @VibhorChinda @sayantani11 :)

Reply:
Absolutely. My idea was to give some ideas here and then it's up to the first coms team to get a consensus and update the document accordingly

Reply:
cc @joekr

Reply:
Added it as "Define details of "Communicate key dates to the community"" to improvement tasks in https://github.com/kubernetes-sigs/cluster-api/issues/7389

@oscr As #7389 and the issue we will move the improvement tasks to are only meant as umbrella issues, feel free to open a specific issue for defining the details of this task.

I think it would be good to decouple this discussion from this PR